cardano-crypto-class/src/Cardano/Crypto/KES/CompactSingle.hs

```diff
@@ -168,6 +168,39 @@ instance ( DSIGNMAlgorithm d
     forgetSignKeyKESWith allocator (SignKeyCompactSingleKES v) =
       forgetSignKeyDSIGNMWith allocator v
 
+instance ( KESAlgorithm (CompactSingleKES d)
+         , UnsoundDSIGNMAlgorithm d
+         )
+         => UnsoundPureKESAlgorithm (CompactSingleKES d) where
+    data UnsoundPureSignKeyKES (CompactSingleKES d) =
+              UnsoundPureSignKeyCompactSingleKES (SignKeyDSIGN d)
+        deriving (Generic)
+
+    unsoundPureSignKES ctxt t a (UnsoundPureSignKeyCompactSingleKES sk) =
+        assert (t == 0) $!
+        SigCompactSingleKES (signDSIGN ctxt a sk) (deriveVerKeyDSIGN sk)
+
+    unsoundPureUpdateKES _ctx _sk _to = Nothing
+
+    --
+    -- Key generation
+    --
+
+    unsoundPureGenKeyKES seed =
+      UnsoundPureSignKeyCompactSingleKES $! genKeyDSIGN seed
+
+    unsoundPureDeriveVerKeyKES (UnsoundPureSignKeyCompactSingleKES v) =
+      VerKeyCompactSingleKES $! deriveVerKeyDSIGN v
+
+    unsoundPureSignKeyKESToSoundSignKeyKES (UnsoundPureSignKeyCompactSingleKES sk) =
+      maybe (error "unsoundPureSignKeyKESToSoundSignKeyKES: deserialisation failure") (return . SignKeyCompactSingleKES)
+      =<< (rawDeserialiseSignKeyDSIGNM . rawSerialiseSignKeyDSIGN $ sk)
```

Comment:
Redundant composition
```suggestion
      =<< (rawDeserialiseSignKeyDSIGNM $ rawSerialiseSignKeyDSIGN sk)
```

Reply:
I like to highlight the way deserialize and serialize pair up here, hence the composition style. Not super opinionated about it though.